db/data_migration/20220812212927_backfill_latest_edition_and_live_edition_on_document.rb

```diff
@@ -0,0 +1,12 @@
+# Populate latest_edition_id and live_edition_id for documents that don't have them defined
+all_documents = Document.where(latest_edition_id: nil, live_edition_id: nil)
+total = all_documents.count
+
+@logger.info "There are #{total} documents to populate"
+
+done = 0
+all_documents.find_in_batches do |documents|
+  documents.each(&:update_edition_references!)
+  done += documents.count
+  @logger.info "Done #{done}/#{total} (#{((done / total.to_f) * 100).to_i}%)"
+end
```

Comment:
I don't think it's a big deal, but this will be pretty slow right? I'm guessing it'll take a good few minutes.

Is there a downside to doing something like this?

```
execute <<-SQL
      UPDATE documents
      SET live_edition_id = (
        SELECT MAX(id)
        FROM editions
        WHERE documents.id = editions.document_id
        WHERE documents.live_edition_id IS NULL
        AND editions.state IN ('published', 'withdrawn')
      )
    SQL
```

Reply:
Yea all fair points! It takes about 45 seconds to run the raw SQL on the int dataset though so it's quite a bit quicker.

Reply:
Oh blimey 🙈 that's WAY quicker!!

Reply:
The downside of a single SQL query is that you can lock all the records during the transaction and thus not allow anyone to write to the DB until it's complete. I'm not too sure about how that plays out in MySQL but it can be an annoying for Postgres.

Though if you're talking 45s vs 1 hour I'd probably go with the fast query.

Reply:
Not to dismiss this suggestion, but I think I'm going to leave the database migration as is since neither of us seem to hold a strong opinion either way.

However, the SQL query may prove handy on dev machines if we want to quickly backfill data without pulling down a fresh copy of the integration database.